drivers/sensor/bosch/bmm350/bmm350.c

```diff
@@ -0,0 +1,984 @@
+		k_usleep(BMM350_BR_DELAY);
+	}
+
+	if (ret == BMM350_OK) {
+		/* Verify if PMU_CMD_STATUS_0 register has BR set */
+		ret = bmm350_get_pmu_cmd_status_0(&pmu_cmd_stat_0, dev);
+		LOG_DBG("get status result 1:%d\n", ret);
+		if ((ret == BMM350_OK) &&
+		    (pmu_cmd_stat_0.pmu_cmd_value != BMM350_PMU_CMD_STATUS_0_BR)) {
+			ret = BMM350_E_PMU_CMD_VALUE;
+		}
+	}
+
+	if (ret == BMM350_OK) {
+		/* Set FGR to PMU_CMD register */
+		ret = bmm350_reg_write(dev, BMM350_REG_PMU_CMD, BMM350_PMU_CMD_FGR);
+		k_usleep(BMM350_FGR_DELAY);
+	}
+	if (ret == BMM350_OK) {
+		/* Verify if PMU_CMD_STATUS_0 register has FGR set */
+		ret = bmm350_get_pmu_cmd_status_0(&pmu_cmd_stat_0, dev);
+		LOG_DBG("get status result 2:%d\n", ret);
+
+		if ((ret == BMM350_OK) &&
+		    (pmu_cmd_stat_0.pmu_cmd_value != BMM350_PMU_CMD_STATUS_0_FGR)) {
+			ret = BMM350_E_PMU_CMD_VALUE;
+		}
+	}
+
+	if ((ret == BMM350_OK) && (restore_normal == BMM350_ENABLE)) {
+		ret = bmm350_set_powermode(BMM350_NORMAL_MODE, dev);
+		LOG_DBG("set power mode 1:%d\n", ret);
+	} else {
+		if (ret == BMM350_OK) {
+			/* Reset PMU_CMD register */
+			ret = bmm350_reg_write(dev, BMM350_REG_PMU_CMD, 0x00);
+		}
+	}
+	return ret;
+}
+/*!
+ * @brief This API is used to read uncompensated mag and temperature data.
+ */
+int8_t bmm350_read_uncomp_mag_temp_data(struct bmm350_raw_mag_data *raw_data,
+					const struct device *dev)
+{
+	struct bmm350_data *data = dev->data;
+	int8_t rslt = BMM350_OK;
+	uint8_t mag_data[14] = {0};
+	uint32_t raw_mag_x, raw_mag_y, raw_mag_z, raw_temp;
+
+	if (raw_data != NULL) {
+		/* Get uncompensated mag data */
+		rslt = bmm350_reg_read(dev, BMM350_REG_MAG_X_XLSB, mag_data, 14);
+
+		if (rslt == BMM350_OK) {
+			raw_mag_x = (uint32_t)mag_data[2] + ((uint32_t)mag_data[3] << 8) +
+				    ((uint32_t)mag_data[4] << 16);
+			raw_mag_y = (uint32_t)mag_data[5] + ((uint32_t)mag_data[6] << 8) +
+				    ((uint32_t)mag_data[7] << 16);
+			raw_mag_z = (uint32_t)mag_data[8] + ((uint32_t)mag_data[9] << 8) +
+				    ((uint32_t)mag_data[10] << 16);
+			raw_temp = (uint32_t)mag_data[11] + ((uint32_t)mag_data[12] << 8) +
+				   ((uint32_t)mag_data[13] << 16);
+
+			if ((data->axis_en & BMM350_EN_X_MSK) == BMM350_DISABLE) {
+				raw_data->raw_xdata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_xdata = fix_sign(raw_mag_x, BMM350_SIGNED_24_BIT);
+			}
+
+			if ((data->axis_en & BMM350_EN_Y_MSK) == BMM350_DISABLE) {
+				raw_data->raw_ydata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_ydata = fix_sign(raw_mag_y, BMM350_SIGNED_24_BIT);
+			}
+
+			if ((data->axis_en & BMM350_EN_Z_MSK) == BMM350_DISABLE) {
+				raw_data->raw_zdata = BMM350_DISABLE;
+			} else {
+				raw_data->raw_zdata = fix_sign(raw_mag_z, BMM350_SIGNED_24_BIT);
+			}
+
+			raw_data->raw_data_t = fix_sign(raw_temp, BMM350_SIGNED_24_BIT);
+		}
+	} else {
+		rslt = BMM350_E_NULL_PTR;
+	}
+
+	return rslt;
+}
+static int8_t read_out_raw_data(int32_t *out_data, const struct device *dev)
```

Comment:
nit: usually, I always see the `const struct device *` as the first arg for every function. Just something that seems to not follow a pattern